x-pack/plugins/enterprise_search/public/applications/shared/telemetry/send_telemetry.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext, useEffect } from 'react';
+
+import { HttpHandler } from 'src/core/public';
+import { KibanaContext, IKibanaContext } from '../../index';
+
+interface ISendTelemetryProps {
+  action: 'viewed' | 'error' | 'clicked';
+  value: string; // e.g., 'setup_guide'
+}
+
+interface ISendTelemetry extends ISendTelemetryProps {
+  http(): HttpHandler;
+  product: 'app_search' | 'workplace_search' | 'enterprise_search';
+}
+
+/**
+ * Base function - useful for non-component actions, e.g. clicks
+ */
+
+export const sendTelemetry = ({ http, product, action, metric }: ISendTelemetry) => {
+  return http
+    .put(`/api/${product}/telemetry`, {
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({ action, metric }),
+    })
+    .catch(e => {
+      throw e;
```

Comment:
What are we catching this if we're just throwing it again?

Reply:
Ah shoot thanks, I accidentally left this in here from debugging. I actually don't think we're supposed to be throwing raw errors to the console per Kibana's security recommendations, but instead some kind of safe string like 'Couldn't record telemetry'. IIRC it's so we don't expose sensitive information by accident in stack traces.

Which reminds me though that I think what we should do is log errors to via Kibana's logger plugin 🤔

Reply:
Updated: https://github.com/constancecchen/kibana/pull/5/files#diff-5bf73fdb46d840f67b85019bef1e0262R33